lib/df-detection.c

```diff
@@ -0,0 +1,275 @@
+#include "linux/df-detection.h"
```

Comment:
Your header lives in include/linux as well as the ones below. Please insert it between them (and change quotes to angle brackets)